### DISCUSSION TITLE
how to check shell command if exists?

### DISCUSSION THREAD

**Discussion by skanehira** [denoland/deno#10654] [Q&A]
I want to check shell command if exists with deno but there is no way?

In my mind is bellow.
```ts
if (executable("vim")) { // do something }
```

I try to found functions but I couldn't found it form modules.

https://deno.land/std@0.96.0/path
https://deno.land/std@0.96.0/fs

**Comment by bnoordhuis** [comment]
I've moved this to Discussions. I think the short answer to your question is "don't do that" - just try to execute the command and handle the failure when it doesn't exist. Any other approach is vulnerable to TOCTOU bugs.
